DESIGN.md

```diff
@@ -33,26 +33,22 @@ During the bootstrap of OpenSearch node, it class loads all the code under `~/pl
 ![](Docs/Extensions.png)
 
 Extensions are independent processes which are built using `opensearch-sdk`. They communicate with OpenSearch via [transport](https://github.com/opensearch-project/OpenSearch/tree/main/modules/transport-netty4) protocol which today is used to communicate between OpenSearch nodes.  
-Extensions are designed to extend features via APIs which are exposed using extension points of OpenSearch.  
+Extensions are designed to extend features via transport APIs which are exposed using extension points of OpenSearch.
 
 ### Discovery
-Extensions are discovered and configured via `extensions.yml` which is read by OpenSearch during the node bootstrap. `ExtensionsOrchestrator` reads through the config file at `~/extensions` and registers extensions within OpenSearch.
+Extensions are discovered and configured via `extensions.yml`, same way we currently have `plugin-descriptor.properties` which is read by OpenSearch during the node bootstrap. `ExtensionsOrchestrator` reads through the config file at `~/extensions` and registers extensions within OpenSearch.
 Here is an example extension configuration `extensions.yml`:
 
 ```
 extensions:
-  - name: opensearch-sdk
-    uniqueId: opensearch-sdk-1
-    hostName: 'sdk_host'
-    hostAddress: '127.0.0.1'
-    port: '4532'
-    version: '1.0'
-    description: Extension for the Opensearch SDK Repo
-    opensearchVersion: '3.0.0'
-    javaVersion: '14'
-    className: ExtensionsRunner
-    customFolderName: opensearch-sdk
-    hasNativeController: false	
+  - name: opensearch-sdk // extension name
+    uniqueId: opensearch-sdk-1 // identifier for the extension
+    hostName: 'sdk_host' // name of the host where extension is running
+    hostAddress: '127.0.0.1' // host to reach
```

Comment:
Why both, shouldn't the name be resolved into the address?

Reply:
Ideally yes. As we are using the existing data structure `DiscoveryNode` [1]. 
We didnt really want to redefine everything Transport. 

[1] https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/cluster/node/DiscoveryNode.java#L126-L134